### PR TITLE
Fix the Fedora 36 build

### PR DESCRIPTION
**Pull request by fwiesel**
- Update mkosi and requirements
- Backport patch from main (for new gpg keys)
- mkosi now reads mkosi.conf (and mkosi.conf.d) instead of mkosi.files
- Chmod the generated directory so the owner can actually modify it
- `systemctl` commands do not work in `mkosi.postinst` anymore, move the commands to `mkosi.finalize` with pointing systemctl to the root
- Git is now more strict, so explicitly add the checked out repositories to the list of safe directories

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "mkosi"]
 	path = mkosi
-	url = https://github.com/systemd/mkosi.git
+	url = https://github.com/sapcc/mkosi.git
 	branch = main
```
Comment (sandzwerg) [inline .gitmodules:4]: If you want to include your patch I think you need to use branch v14, there I see a commit from you but on main I don't

**Review by sandzwerg — CHANGES_REQUESTED** [review]
Apart from the wrong branch looks good to me. After that is done, we might try to use a newer fedora version so we don't go out of support

File: .gitmodules (hunk 1 above)
Comment (fwiesel) [reply .gitmodules]: Fixed it

**Comment by sandzwerg** [comment]
hmmkay then we need to look into that, but first let's do this.

**Review by sandzwerg — APPROVED** [review]

**Pull request merged**